lib/src/location_status_adapter.dart

```diff
@@ -0,0 +1,81 @@
+part of sdk;
+
+/// [LocationStatusAdapter]
+/// This class is in charge of handling the native LocationStatus
+/// that we are recieving from the native SDKs (Android and iOS),
+/// and giving a common status and behaviour between platforms.
+class LocationStatusAdapter {
+  late bool _isAndroid;
+  String? _lastStatus;
+
+  LocationStatusAdapter() {
+    _isAndroid = Platform.isAndroid;
+  }
+
+  String? handleStatus(String status) {
+    return _isAndroid ? _handleAndroidStatus(status) : _handleIOSStatus(status);
+  }
+
+  // Native Android statuses
+  String? _handleAndroidStatus(String status) {
+    String? parsedStatus;
+    switch (status) {
+      case "CALCULATING":
+      case "USER_NOT_IN_BUILDING":
+        if (shouldNotifyStatus(status)) {
+          parsedStatus = status;
+          _lastStatus = parsedStatus;
+        }
+        break;
+      case "AUTO_ENABLE_BLE_FORBIDDEN":
+      case "COMPASS_CALIBRATION_NEEDED":
+      case "COMPASS_CALIBRATION_NOT_NEEDED":
+      case "WIFI_SCAN_THROTTLED":
+      case "TIME_SETTINGS_MANUAL":
+      case "LOCATION_DISABLED":
+      case "BLE_SENSOR_DISABLED_BY_USER":
+      case "BLE_NOT_AVAILABLE":
+      case "ALARM_PERMISSIONS_NEEDED_TO_AVOID_DOZE":
+      case "GEOFENCES_NOT_AVAILABLE":
+      case "GLOBAL_LOCATION_NOT_FOUND":
+      case "STOPPED":
+        parsedStatus = status;
```

Comment:
Por que nestos casos non se asigna _lastStatus?

Reply:
porque _lastStatus e unha variable para non repetir nin USER_NOT_IN_BUILDING e STOPPED. É unha variable auxialiar.

Reply:
Non tería más sentido poñer un default para asignar o parsedStatus = status en lugar de listar todolos estados, se queres excluir un, pois pon o case do que exclues e non o asignes, pero creo que che quedaría máis sinxelo.

Entendo que o lastStatus tes que asignalo sempre, non, senon non sabes se houbo outro estado polo medio que che resetee a conta

Dito esto non me complicaría co de que se repitan estados en Android si e en iOS non. Creo que o importante é que sexan coherentes en orde

Reply:
Cambiei a loxica seguindo as tuas indicacions, ao final acabo antes ignorando os estados que non quero. En canto ao da repeticion dos CALCULATING e USER_NOT_IN_BUILDING, metin un metodo para nulear o _lastStatus cando nos chegan locations para resetear o estado